File: server/src/main/scala/kpn/api/custom/RouteTypeLetter.scala
package kpn.api.custom

import kpn.api.common.RouteType

object RouteTypeLetter {

  def letter(routeType: RouteType): String = {
    routeType match {
      case RouteType.hiking => "w"
      case RouteType.cycling => "c"
      case RouteType.horseRiding => "h"
      case RouteType.canoe => "p"
      case RouteType.motorboat => "m"
      case RouteType.inlineSkating => "i"
      case RouteType.mtb => "?"
    }
  }
}

File: backend/api/src/main/java/kpn/api/custom/RouteTypeLetter.java
package kpn.api.custom;

import kpn.api.common.RouteType;

public class RouteTypeLetter {

  public static String letter(RouteType routeType) {
    switch (routeType) {
      case RouteType.HIKING:
        return "w";
      case RouteType.CYCLING:
        return "c";
      case RouteType.HORSE_RIDING:
        return "h";
      case RouteType.CANOE:
        return "p";
      case RouteType.MOTORBOAT:
        return "m";
      case RouteType.INLINE_SKATING:
        return "i";
      case RouteType.MTB:
        return "?";
      default:
        throw new IllegalArgumentException("Unknown RouteType");
    }
  }
}
